### PR TITLE
functions filesystem: partitions_add: fix number of required arguments bug

### PR DESCRIPTION
**Pull request by stdweird**
Code clean-up makes use of pan 10.7 features.

Will need https://github.com/quattor/scdb/pull/46 to be merged to pass Travis tests.

### CODE REVIEW

**Comment by stdweird** [comment]
this requires 10.7 compiler

**Comment by jrha** [comment]
Hang on, this *requires* panc 10.7?
If that is the case this makes me very uncomfortable, I thought this just allowed the templates to work with 10.7.

**Comment by stdweird** [comment]
to make it work with 10.7, it's only 2nd commit https://github.com/quattor/template-library-core/pull/174/commits/bc972a7e96500daf1b589d72741be9422f5d3bb1
but the cleanup uses features from 10.7

so, yeah, i should better make a PR that just fixes the `null`/undef` issue

but if panlint fails, and you want me to fix it, i won't unless i can use 10.7....

**Comment by stdweird** [comment]
i made #175 to allow transition to 10.7

**Comment by jrha** [comment]
Sure, thanks!

**Comment by jrha** [comment]
@stdweird can you rebase and fix the conflicts?

File: quattor/functions/filesystem.pan [hunk 1]
```diff
@@ -1,58 +1,53 @@
...
-      return(i);
+    base = "/hardware/harddisks";
+    dsk = value(base);
+    foreach (i; v; dsk) {
+        path = format("%s/%s/boot", base, i);
+        if ( exists(to_string(path)) && value(path) ) {
+            return(i);
+        };
     };
   };
```
Comment (jrha) [inline quattor/functions/filesystem.pan]: This `};` shouldn't be here.

**Pull request merged**